lib/plugins/nextcloud-vue/rules/no-deprecated-props.test.ts

```diff
@@ -0,0 +1,72 @@
+/*!
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+import { RuleTester } from '@typescript-eslint/rule-tester'
+import { afterAll, describe, test } from 'vitest'
+import vueParser from 'vue-eslint-parser'
+import rule from './no-deprecated-props.ts'
+
+RuleTester.afterAll = afterAll
+RuleTester.describe = describe
+RuleTester.it = test
+
+const ruleTester = new RuleTester({
+	languageOptions: {
+		parser: vueParser,
+		ecmaVersion: 2015,
+	},
+})
+
+ruleTester.run('no-deprecated-props', rule, {
+	valid: [
+		{
+			code: '<template><NcButton variant="primary">Hello</NcButton></template>',
+			filename: 'test.vue',
+		},
+		{
+			code: '<template><NcButton type="submit">Hello</NcButton></template>',
+			filename: 'test.vue',
+		},
+		{
+			code: '<template><NcButton variant="tertiary-no-background" type="reset">Hello</NcButton></template>',
+			filename: 'test.vue',
+		},
+		{
+			code: '<template><NcButton :type="isReset ? \'reset\' : \'submit\'">Hello</NcButton></template>',
+			filename: 'test.vue',
+		},
+	],
+	invalid: [
+		{
+			code: '<template><NcButton type="primary">Hello</NcButton></template>',
+			filename: 'test.vue',
+			errors: [{ messageId: 'useVariantInstead' }],
+			output: '<template><NcButton variant="primary">Hello</NcButton></template>',
+		},
+		{
+			code: '<template><NcButton type="tertiary">Hello</NcButton></template>',
+			filename: 'test.vue',
+			errors: [{ messageId: 'useVariantInstead' }],
+			output: '<template><NcButton variant="tertiary">Hello</NcButton></template>',
+		},
+		{
+			code: '<template><NcButton type="error">Hello</NcButton></template>',
+			filename: 'test.vue',
+			errors: [{ messageId: 'useVariantInstead' }],
+			output: '<template><NcButton variant="error">Hello</NcButton></template>',
+		},
+		{
+			code: '<template><NcButton native-type="button">Hello</NcButton></template>',
+			filename: 'test.vue',
+			errors: [{ messageId: 'useTypeInstead' }],
+			output: '<template><NcButton type="button">Hello</NcButton></template>',
+		},
+		{
```

Comment:
To check a case with both props
```suggestion
		},
		{
			code: '<template><NcButton type="primary" native-type="button">Hello</NcButton></template>',
			filename: 'test.vue',
			errors: [{ messageId: 'useTypeInstead' }, { messageId: 'useVariantInstead' }],
			output: '<template><NcButton variant="primary" type="button">Hello</NcButton></template>',
		},
		{
```